src/runtime/meter-toast-api.js

```diff
@@ -37,18 +37,23 @@ export const DEFAULT_IFRAME_URL = '/metertoastiframe';
  * - iframeUrlParams: List of extra params appended to the URL.
  *
  * @typedef {{
- *   iframeUrl: (string|undefined),
- *   iframeUrlParams: (?Object<string, string>|undefined),
+ *   iframeUrl: (string),
+ *   iframeUrlParams: (Object<string, string>),
  * }}
  */
 export let MeterToastApiParams;
 
 export class MeterToastApi {
   /**
    * @param {!./deps.DepsDef} deps
-   * @param {?MeterToastApiParams=} params
+   * @param {!MeterToastApiParams=} params
+   * @param {string} params.iframeUrl
```

Comment:
Hmmmm it seems pretty weird that you have to specify these as params here even though they are described in the typedef. Does it still work if you remove these? I think you're also saying the params are optional but not nullable?
It seems like you should just be able to have

```
@param {MeterToastApiParams=} params
```

or you could also just inline the type since I don't think it needs to be referenced

```
/**
   * @param {!./deps.DepsDef} deps
   * @param {{
       *   iframeUrl: (string),
       *   iframeUrlParams: (Object<string, string>),
   }=} params
*/
constructor(...)
```

Reply:
> Hmmmm it seems pretty weird that you have to specify these as params here even though they are described in the typedef.

Yeah, I was following the guideline [here](https://jsdoc.app/tags-param.html#parameters-with-properties) since the [doc](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System) for Closure type system doesn't seem to mention how to document a destructuring param. But when I switched to simply use `@param {!MeterToastApiParams=} params` the type check doesn't complain so I think that's okay.